tf/edgeml/trainer/bonsaiTrainer.py

```diff
@@ -389,18 +418,59 @@ def train(self, batchSize, totalEpochs, sess,
 
             # This helps in direct testing instead of extracting the model out
 
-            testAcc, testLoss, regTestLoss = sess.run(
-                [self.accuracy, self.loss, self.regLoss], feed_dict=_feed_dict)
+            testAcc, testLoss, regTestLoss, pred = sess.run(
+                [self.accuracy, self.loss, self.regLoss, self.prediction], feed_dict=_feed_dict)
+
             if ihtDone == 0:
-                maxTestAcc = -10000
-                maxTestAccEpoch = i
-            else:
-                if maxTestAcc <= testAcc:
+                if (self.bonsaiObj.isRegression == False):
+                    maxTestAcc = -10000
                     maxTestAccEpoch = i
+                elif (self.bonsaiObj.isRegression == True):
                     maxTestAcc = testAcc
```

Comment:
For regression initialize maxTestAcc to very high value and state that you are using this for MAE

Reply:
But aren't we interested only in the minimum MAE in the end.

Reply:
@siddharthdivi , Bonsai needs to be within the stipulated budget, we don't want to take the best model if it is from the dense phase. Get it?